SofaKernel/framework/sofa/core/objectmodel/Base.cpp

```diff
@@ -439,6 +446,33 @@ bool Base::parseField( const std::string& attribute, const std::string& value)
         {
             if (!dataVec[d]->setParent(value))
             {
+                BaseData* data = NULL;
```

Comment:
Can you use `shared_ptr` or `unique_ptr`, and set default to `nullptr` instead of `NULL`